src/proxy/inbound.rs

```diff
@@ -385,23 +378,23 @@ impl Inbound {
     }
 
     async fn check_waypoint(
-        workloads: &WorkloadInformation,
+        state: &DemandProxyState,
```

Comment:
nit: I noticed in some places we use `DemandProxyState` and some `&DemandProxyState`.

Cloning should be cheap. All functions take `&self` so reference is sufficient. So I guess we just use `DemandProxyState` if we don'rt want to deal with lifetimes and `&DemandProxyState` otherwise?

I am not sure if there is a standard practice here